hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsHttpOperation.java

```diff
@@ -320,6 +320,11 @@ public void sendRequest(byte[] buffer, int offset, int length) throws IOExceptio
       // accompanying statusCode
       this.bytesSent = length;
       outputStream.write(buffer, offset, length);
+    } catch (IOException e) {
+      // If getOutputStream fails with an exception due to 100-continue
+      // enabled, we update the bytes sent before they are sent
+      // in the catch block.
+      this.bytesSent = length;
```

Comment:
this is going to swallow all IOEs raised. unless i've misreasd something, the IOE *must* be rethrown